relay/server.go

```diff
@@ -86,23 +97,29 @@ func NewServer(
 	}
 
 	return &Server{
+		config:           config,
 		logger:           logger,
 		grpcPort:         config.GRPCPort,
 		maxProtoSize:     config.MaxGRPCMessageSize,
 		metadataProvider: mp,
 		blobProvider:     bp,
 		chunkProvider:    cp,
+		blobRateLimiter:  limiter.NewBlobRateLimiter(&config.RateLimits),
+		chunkRateLimiter: limiter.NewChunkRateLimiter(&config.RateLimits),
 	}, nil
 }
 
 // GetBlob retrieves a blob stored by the relay.
 func (s *Server) GetBlob(ctx context.Context, request *pb.GetBlobRequest) (*pb.GetBlobReply, error) {
 
-	// Future work	:
-	//  - global throttle
-	//  - per-connection throttle
+	// Future work:
 	//  - timeouts
 
+	err := s.blobRateLimiter.BeginGetBlobOperation(time.Now())
```

Comment:
Yes, this was missing. Passed unit tests because the default concurrency limit is higher than the maximum number of operations executed by any single unit test on the server. Fixed.